internal/fleet/agent_policy/schema.go

```diff
@@ -12,74 +12,86 @@ import (
+			},
+			"monitor_logs": schema.BoolAttribute{
+				Description: "Enable collection of agent logs.",
+				Computed:    true,
+				Optional:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"monitor_metrics": schema.BoolAttribute{
+				Description: "Enable collection of agent metrics.",
+				Computed:    true,
+				Optional:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"skip_destroy": schema.BoolAttribute{
+				Description: "Set to true if you do not wish the agent policy to be deleted at destroy time, and instead just remove the agent policy from the Terraform state.",
+				Optional:    true,
 			},
-		},
-		"policy_id": schema.StringAttribute{
-			Description: "Unique identifier of the agent policy.",
-			Computed:    true,
-			Optional:    true,
-			PlanModifiers: []planmodifier.String{
-				stringplanmodifier.UseStateForUnknown(),
-				stringplanmodifier.RequiresReplace(),
+			"sys_monitoring": schema.BoolAttribute{
+				Description: "Enable collection of system logs and metrics.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
+				},
 			},
-		},
-		"name": schema.StringAttribute{
-			Description: "The name of the agent policy.",
-			Required:    true,
-		},
-		"namespace": schema.StringAttribute{
-			Description: "The namespace of the agent policy.",
-			Required:    true,
-		},
-		"description": schema.StringAttribute{
-			Description: "The description of the agent policy.",
-			Optional:    true,
-		},
-		"data_output_id": schema.StringAttribute{
-			Description: "The identifier for the data output.",
-			Optional:    true,
-		},
-		"monitoring_output_id": schema.StringAttribute{
-			Description: "The identifier for monitoring output.",
-			Optional:    true,
-		},
-		"fleet_server_host_id": schema.StringAttribute{
-			Description: "The identifier for the Fleet server host.",
-			Optional:    true,
-		},
-		"download_source_id": schema.StringAttribute{
-			Description: "The identifier for the Elastic Agent binary download server.",
-			Optional:    true,
-		},
-		"monitor_logs": schema.BoolAttribute{
-			Description: "Enable collection of agent logs.",
-			Computed:    true,
-			Optional:    true,
-			Default:     booldefault.StaticBool(false),
-		},
-		"monitor_metrics": schema.BoolAttribute{
-			Description: "Enable collection of agent metrics.",
-			Computed:    true,
-			Optional:    true,
-			Default:     booldefault.StaticBool(false),
-		},
-		"skip_destroy": schema.BoolAttribute{
-			Description: "Set to true if you do not wish the agent policy to be deleted at destroy time, and instead just remove the agent policy from the Terraform state.",
-			Optional:    true,
-		},
-		"sys_monitoring": schema.BoolAttribute{
-			Description: "Enable collection of system logs and metrics.",
-			Optional:    true,
-			PlanModifiers: []planmodifier.Bool{
-				boolplanmodifier.RequiresReplace(),
+			"global_data_tags": schema.StringAttribute{
```

Comment:
I'm not a big fan of this being an arbitrary json string. It's hard to document and build proper validation around. 

At the very least we should set `CustomType:  jsontypes.NormalizedType{}` and add a validator ensuring the json string can be serialised into `[]kbapi.AgentPolicyGlobalDataTagsItem`.

I do think it would be nicer to use a `MapNestedAttribute` here with an element like `object({ string_value = optional(string), number_value = optional(number) })`. IMO it would provide a much nicer provider first user experience rather than having to dive through both provider and Fleet documentation. Language server users would also get proper autocompletion/inline linting as well.